### ISSUE TITLE
Why the book isn't building

### ISSUE THREAD

**Issue by jennybc** [hadley/r-pkgs#713]
Want to record my thoughts on this until I resolve it, which will be easier once I get through some local computer updating (e.g. once I'm on R 4.1).

It seems to be this call to `devtools::build_readme()`:

https://github.com/hadley/r-pkgs/blob/f410975b072559b0b8e7832b90617013f68ebf0e/Whole-game.Rmd#L745

Ultimately that calls `devtools::install()` on the local package (foofactors, or whatever our practice package may be) and it seems likely to be the same problem as I saw @gaborcsardi diagnose for someone else on slack recently. Cc @jimhester.

> This is an incompatibility between pak and remotes, and an unknown package type throws an error now in R 4.1:
``` r
❯ available.packages(type = "macos")
Error in contrib.url(repos, type) : invalid 'type'
```

So perhaps this comes from the overall book-building workflow using pak, but devtools still relying on `remotes::install_deps()` (via `build_readme()` --> `build_rmd()` --> `install()`).

**Comment by gaborcsardi** [comment]
> it seems likely to be the same problem as I saw @gaborcsardi diagnose for someone else on slack recently

Yes, that seems like a clash between remotes and pak, and one workaround is to use pak only and bump the cache. It will also be fixed soon: https://github.com/r-lib/remotes/issues/617

**Comment by jennybc** [comment]
Well, in this case, the clashing use of pak vs remotes is not within my control. I'm using pak for the book-building, but devtools still makes some use of remotes, which gets called by the *content* of the book.

**Comment by gaborcsardi** [comment]
Oh, right, I'll fix it soon, then:
https://github.com/r-lib/pak/issues/301

**Comment by gaborcsardi** [comment]
@jennybc Should be fixed now, I built a new pak binary for R-4.1 macOS, which the book seems to be using. For other platforms it will be fixed tomorrow.

You'll need to bump the cache. Let me know if you still see the error.

**Comment by jennybc** [comment]
Yep, seems to be resolved now! Thanks @gaborcsardi.
